cpp/src/arrow/pretty_print.cc

```diff
@@ -232,9 +242,9 @@ class ArrayPrinter : public PrettyPrinter {
   enable_if_has_string_view<T, Status> WriteDataValues(const ArrayType& array) {
     return WriteValues(array, [&](int64_t i) {
       if constexpr (T::is_utf8) {
-        (*sink_) << "\"" << array.GetView(i) << "\"";
+        this->Write("\"" + std::string{array.GetView(i)} + "\"");
```

Comment:
We could perhaps avoid the string copy by turning:
```c++
void PrettyPrinter::Write(std::string_view data);
```
into
```c++
void PrettyPrinter::Write(std::string_view data, int max_chars);
```
and then changing this snippet into:
```suggestion
        (*sink_) << "\"";
        this->Write(array.GetView(i), options_.element_size_limit - 2);
        (*sink_) << "\"";
```

Another possibility is to let `Write` take an arbitrary number of arguments and then:
```suggestion
        this->Write("\"", array.GetView(i), "\"");
```
```